### ISSUE TITLE
Multiple schema, enums not generated for second one

### ISSUE THREAD

**Issue by kono0514** [comigor/artemis#199]
**Bug description**
I wanted to consume two separate GraphQL API servers in my client. One is mine and other one is third-party.
These schemas have no connection with each other. Is this even something that is supported in artemis?

Build runs successfully and generates separate ".graphql.dart" files for each schema specified in schema_mapping.
But the enums are only generated for the first schema and not the other one(s), thus throwing:
```dart
// File: graphql_repositories_api.graphql.dart

Undefined class 'Privacy'.
Undefined class 'Status'.
The name 'NotificationOptionInput' isn't a type so it can't be used as a type argument.
```

**Specs**
Artemis version: 6.7.1-beta.1
<details>
<summary>build.yaml:</summary>
  
```yaml
targets:
  $default:
    sources:
      - lib/**
      - graphql/**
      - schema.graphql
      - otherschema.graphql
    builders:
      artemis:
        options:
          schema_mapping:
            - schema: schema.graphql
              queries_glob: graphql/articles.graphql
              output: lib/graphql_api.graphql.dart
            - schema: otherschema.graphql
              queries_glob: graphql/repositories.graphql
              output: lib/graphql_repositories_api.graphql.dart
```
</details>
<details>
<summary>Artemis output:</summary>
  
```bash
[ +266 ms] executing: [C:\flutter\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +151 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] 2ae34518b87dd891355ed6c6ea8cb68c4d52bb9d
[   +1 ms] executing: [C:\flutter\flutter/] git tag --contains HEAD
[ +295 ms] Exit code 0 from: git tag --contains HEAD
[   +1 ms] 1.20.1
[  +16 ms] executing: [C:\flutter\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +52 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [C:\flutter\flutter/] git ls-remote --get-url origin
[  +41 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +116 ms] executing: [C:\flutter\flutter/] git rev-parse --abbrev-ref HEAD
[  +74 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] stable
[ +104 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +6 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +8 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +5 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +29 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +8 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +10 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +5 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[  +10 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +4 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +4 ms] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[  +50 ms] Using C:\flutter\flutter\.pub-cache for the pub cache.
[   +2 ms] executing: C:\flutter\flutter\bin\cache\dart-sdk\bin\pub.bat run build_runner build --delete-conflicting-outputs --verbose
[INFO] Generating build script...
[INFO] Generating build script completed, took 662ms

[WARNING] build.fallback:
The package `flutter_artemis_test` does not include some required sources in any of its targets (see their build.yaml file).
The missing sources are:
  - $package$
[INFO] BuildDefinition:Initializing inputs
[INFO] BuildDefinition:Reading cached asset graph...
[INFO] BuildDefinition:Reading cached asset graph completed, took 177ms

[INFO] BuildDefinition:Checking for updates since last build...
[INFO] BuildDefinition:Checking for updates since last build completed, took 1.5s

[INFO] Build:Running build...
[FINE] artemis:artemis on lib/$lib$:-> Enum
[FINE] artemis:artemis on lib/$lib$:<- Generated enum ArticleType.
[FINE] artemis:artemis on lib/$lib$:-> Class
[FINE] artemis:artemis on lib/$lib$:┌ [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query')][Query][null null] ()
[FINE] artemis:artemis on lib/$lib$:|   [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query')][Query][ClassName(name:r'Article') ClassPropertyName(name:r'articles')] -> List<BrowseArticles$Query$Article>
[FINE] artemis:artemis on lib/$lib$:|   -> Class
[FINE] artemis:artemis on lib/$lib$:|   ┌ [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query'), ClassName(name:r'Article')][Article][null null] ()
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query'), ClassName(name:r'Article')][Article][ClassName(name:r'ID') ClassPropertyName(name:r'id')] -> String
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query'), ClassName(name:r'Article')][Article][ClassName(name:r'String') ClassPropertyName(name:r'title')] -> String
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query'), ClassName(name:r'Article')][Article][ClassName(name:r'ArticleType') ClassPropertyName(name:r'articleType')] -> ArticleType
[FINE] artemis:artemis on lib/$lib$:|   └ [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query'), ClassName(name:r'Article')][Article][null null] ()
[FINE] artemis:artemis on lib/$lib$:|   <- Generated class BrowseArticles$Query$Article.
[FINE] artemis:artemis on lib/$lib$:└ [TypeName(name:r'BrowseArticles'), TypeName(name:r'Query')][Query][null null] ()
[FINE] artemis:artemis on lib/$lib$:<- Generated class BrowseArticles$Query.
[FINE] artemis:artemis on lib/$lib$:-> Class
[FINE] artemis:artemis on lib/$lib$:┌ [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query')][Query][null null] ()
[FINE] artemis:artemis on lib/$lib$:|   [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query')][Query][ClassName(name:r'Repository') ClassPropertyName(name:r'repositories')] -> List<BrowseRepositories$Query$Repository>
[FINE] artemis:artemis on lib/$lib$:|   -> Class
[FINE] artemis:artemis on lib/$lib$:|   ┌ [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query'), ClassName(name:r'Repository')][Repository][null null] ()
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query'), ClassName(name:r'Repository')][Repository][ClassName(name:r'ID') ClassPropertyName(name:r'id')] -> String
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query'), ClassName(name:r'Repository')][Repository][ClassName(name:r'String') ClassPropertyName(name:r'title')] -> String
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query'), ClassName(name:r'Repository')][Repository][ClassName(name:r'Privacy') ClassPropertyName(name:r'privacy')] -> Privacy
[FINE] artemis:artemis on lib/$lib$:|   |   [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query'), ClassName(name:r'Repository')][Repository][ClassName(name:r'Status') ClassPropertyName(name:r'status')] -> Status
[FINE] artemis:artemis on lib/$lib$:|   └ [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query'), ClassName(name:r'Repository')][Repository][null null] ()
[FINE] artemis:artemis on lib/$lib$:|   <- Generated class BrowseRepositories$Query$Repository.
[FINE] artemis:artemis on lib/$lib$:└ [TypeName(name:r'BrowseRepositories'), TypeName(name:r'Query')][Query][null null] ()
[FINE] artemis:artemis on lib/$lib$:<- Generated class BrowseRepositories$Query.
[INFO] Heartbeat:2.5s elapsed, 1/2 actions completed.
[FINE] json_serializable:json_serializable on lib/graphql_repositories_api.graphql.dart:Running JsonSerializableGenerator - 1 of 2
[FINE] json_serializable:json_serializable on lib/graphql_repositories_api.graphql.dart:Running JsonLiteralGenerator - 2 of 2
[INFO] Build:Running build completed, took 5.5s

[INFO] Build:Caching finalized dependency graph...
[INFO] Build:Caching finalized dependency graph completed, took 46ms

[INFO] Build:Succeeded after 5.6s with 5 outputs (5 actions)

[+10889 ms] "flutter run" took 11,066ms.
[ +352 ms] ensureAnalyticsSent: 260ms
[   +3 ms] Running shutdown hooks
[   +5 ms] Shutdown hooks complete
[   +2 ms] exiting with code 0
```
</details>
<details>
<summary>GraphQL schema:</summary>

schema.graphql

```graphql
type Query {
    articles: [Article!]
}

type Article {
  id: ID!
  title: String!
  articleType: ArticleType!
}

enum ArticleType {
  NEWS
  TUTORIAL
}
```

otherschema.graphql

```graphql
type Query {
    repositories(notificationTypes: [NotificationOptionInput]): [Repository!]
}

type Repository {
  id: ID!
  title: String!
  privacy: Privacy!
  status: Status!
}

enum Privacy {
  PRIVATE
  PUBLIC
}

enum Status {
  ARCHIVED
  NORMAL
}

input NotificationOptionInput {
  type: NotificationType
  enabled: Boolean
}

enum NotificationType {
  ACTIVITY_MESSAGE
  ACTIVITY_REPLY
  FOLLOWING
  ACTIVITY_MENTION
}
```
</details>
<details>
<summary>GraphQL query:</summary>

graphql/articles.graphql

```graphql
query BrowseArticles {
    articles {
        id
        title
        articleType
    }
}
```

graphql/repositories.graphql

```graphql
query BrowseRepositories($notificationTypes: [NotificationOptionInput]) {
    repositories(notificationTypes: $notificationTypes) {
        id
        title
        privacy
        status
    }
}
```
</details>

**Comment by vasilich6107** [comment]
Hi. Use `multiple-schema-fix` temporarily until we merge the fix

**Comment by kono0514** [comment]
Using `multiple-schema-fix` (PR #198) build fails with

> Exception: Type NotificationType was not found in schema.
> Make sure your query is correct and your schema is updated.

**Specs**
Artemis version: multiple-schema-fix
<details>
<summary>build.yaml:</summary>
  
```yaml
targets:
  $default:
    sources:
      - lib/**
      - graphql/**
      - schema.graphql
      - otherschema.graphql
    builders:
      artemis:
        options:
          schema_mapping:
            - schema: schema.graphql
              queries_glob: graphql/articles.graphql
              output: lib/graphql_api.graphql.dart
            - schema: otherschema.graphql
              queries_glob: graphql/repositories.graphql
              output: lib/graphql_repositories_api.graphql.dart
```
</details>
<details>
<summary>Artemis output:</summary>
  
```bash
Precompiling executable...
Precompiled build_runner:build_runner.
[INFO] Generating build script...
[INFO] Generating build script completed, took 516ms

[WARNING] The package `flutter_artemis_test` does not include some required sources in any of its targets (see their build.yaml file).
The missing sources are:
  - $package$
[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 74ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 770ms

[WARNING] Invalidating asset graph due to build script update!
[INFO] Cleaning up outputs from previous builds....
[INFO] Cleaning up outputs from previous builds. completed, took 11ms

[INFO] Generating build script...
[INFO] Generating build script completed, took 123ms

[INFO] Creating build script snapshot......
[INFO] Creating build script snapshot... completed, took 17.4s

[WARNING] The package `flutter_artemis_test` does not include some required sources in any of its targets (see their build.yaml file).
The missing sources are:
  - $package$
[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 1.1s

[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 2ms

[INFO] Running build...
[SEVERE] artemis:artemis on lib/$lib$:

Exception: Type NotificationType was not found in schema.
Make sure your query is correct and your schema is updated.
[INFO] Running build completed, took 800ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 92ms

[SEVERE] Failed after 907ms
pub finished with exit code 1
```
</details>
<details>
<summary>GraphQL schema:</summary>

schema.graphql

```graphql
type Query {
    articles: [Article!]
}

type Article {
  id: ID!
  title: String!
  articleType: ArticleType!
}

enum ArticleType {
  NEWS
  TUTORIAL
}
```

otherschema.graphql

```graphql
type Query {
    repositories(notificationTypes: [NotificationOptionInput]): [Repository!]
}

type Repository {
  id: ID!
  title: String!
  privacy: Privacy!
  status: Status!
}

enum Privacy {
  PRIVATE
  PUBLIC
}

enum Status {
  ARCHIVED
  NORMAL
}

input NotificationOptionInput {
  type: NotificationType
  enabled: Boolean
}

enum NotificationType {
  ACTIVITY_MESSAGE
  ACTIVITY_REPLY
  FOLLOWING
  ACTIVITY_MENTION
}
```
</details>
<details>
<summary>GraphQL query:</summary>

graphql/articles.graphql

```graphql
query BrowseArticles {
    articles {
        id
        title
        articleType
    }
}
```

graphql/repositories.graphql

```graphql
query BrowseRepositories($notificationTypes: [NotificationOptionInput]) {
    repositories(notificationTypes: $notificationTypes) {
        id
        title
        privacy
        status
    }
}
```
</details>

**Comment by vasilich6107** [comment]
Ok, I’ll try to reproduce with your data

**Comment by vasilich6107** [comment]
Applied additional fixed and added test based on your case
https://github.com/comigor/artemis/pull/198/files#diff-b7e255d7d2c05c55d132aee6090ab96e

Try one more time

**Comment by kono0514** [comment]
Tested it and it works! Thanks 😃

**Comment by vasilich6107** [comment]
fixed in `v6.8.2-beta.1`
